File: exercise/php/day08/src/Services/ToyProductionService.php
<?php

declare(strict_types = 1);

namespace Services;

use Domain\Toy;
use Domain\ToyRepository;

class ToyProductionService
{
	private ToyRepository $repository;

	public function __construct(ToyRepository $repository)
	{
		$this->repository = $repository;
	}

	public function assignToyToElf(string $toyName): void
	{
		$toy = $this->repository->findByName($toyName);

		if (null !== $toy && $toy->getState() === Toy::UNASSIGNED) {
			$toy->setState(Toy::IN_PRODUCTION);
			$this->repository->save($toy);
		}
	}
}

File: exercise/C#/day08/ToyProduction/Services/ToyProductionService.cs
using ToyProduction.Domain;

namespace ToyProduction.Services
{
    public class ToyProductionService(IToyRepository repository)
    {
        public void AssignToyToElf(string toyName)
        {
            var toy = repository.FindByName(toyName);
            if (toy is {State: State.Unassigned})
            {
                toy.State = State.InProduction;
                repository.Save(toy);
            }
        }
    }
}